Integrations/integration-Lastline.yml

```diff
@@ -0,0 +1,293 @@
+    //             Method: 'POST',
+    //             Headers: {'Content-Type': ['application/json']}
+    //         }
+    //         );
+    //     console.log(JSON.stringify(res));
+    //     if (res.StatusCode < 200 || res.StatusCode >= 300) {
+    //         throw 'Failed to ' + queryName + ' , request status code: ' + res.StatusCode + ' and Body: ' + res.Body + '.';
+    //     }
+    //     return JSON.parse(res.Body);
+
+    // }
+    var analysisPoll = function(uuid,timeout,interval) {
+        var res = doReq('POST', '/analysis/get', {uuid: uuid});
+
+        var waited=0;
+        while (!analysisReady(res.obj.data) && timeout>waited) {
+            wait(interval);
+            waited+=interval;
+            res = doReq('POST', 'analysis/get', {uuid: uuid});
+        }
+        return res;
+    }
+    var analysisReady = function (data) {
+        return (data && data.score);
+    }
+    var dataToMd = function(data) {
+        var key = Object.keys(data.analysis_subject)[0];
+        var name = data.analysis_subject[key];
+        var md = '';
+        md += '## Lastline analysis for ' + key + ': **' + name +'**\n';
+        md += '### Score: ' + data.score + "\n";
+        md += 'Task UUID: ' + data.task_uuid + "\n";
+        md += 'Submission Time: ' + data.submission + "\n";
+        if (Array.isArray(data.malicious_activity) && data.malicious_activity.length>0) {
+            md += '### Suspicious Activity\n';
+            md += objToMd(data.malicious_activity) + "\n";
+        }
+        md += "\n";
+        return md;
+    }
+    // var submitFile = function (file,timeout) {
+    //     file = {MD5: '47439da9cdcf7e6379e032334004dc11',
+    //             SHA1: '48818704bca867afd66a5a4d01938ab65d1595c3',
+    //             SHA256: '44403483a46def5191b998b2fd3de809130f100d41b59d57771ad2a9ec7dea52'};
+    //     var res = doReq('POST','analysis/submit/file', {md5: file.MD5}); //, sha1: file.SHA1, sha256: file.SHA256});
+    //     //var res = doFile(file_id);
+    //     var md="";
+    //     var ec={};
+
+    //     return {Type: entryTypes.note, Contents: res.obj, ContentsFormat: formats.json, HumanReadable: md, EntryContext: ec};
+    // }
+    var submitUrl = function (url,timeout,interval) {
+        if (!timeout) {
+            timeout = lastlineDefaults.timeout;
+        }
+        if (!interval) {
+            interval = lastlineDefaults.interval;
+        }
+
+        var res = doReq('POST', 'analysis/submit/url', {url: url});
+        var uuid = res.obj.data.task_uuid;
+
+        var ec = {};
+        var md = '';
+
+        ec.lastlineTasks=[{'url':url,'uuid':uuid}];
+
+        if (!analysisReady(res.obj.data) && timeout > 0) {
+            res = analysisPoll(uuid,timeout,interval);
+        }
+
+        if (analysisReady(res.obj.data)) {
+            md=dataToMd(res.obj.data);
+            if (res.obj.data.score > params.malicious_score) {
+                ec.URLMalicious={'url':url,'score':res.obj.data.score};
```

Comment:
@slavikm - this is based on Lior's document.... need to decide already, so can close this.